site/content/en/docs/Advanced/controlling-disruption.md

```diff
@@ -0,0 +1,115 @@
+
+```
+eviction:
+  safe: Never
+```
+
+Note: If your session length is greater than an hour, Agones may not be able to protect your game server from disruption. See [below](#considerations-for-long-sessions).
+
+### `safe: Always`
+
+If your game server is tolerant of disruption in a timely fashion, you should set:
+
+```
+eviction:
+  safe: Always
+```
+
+You further need to configure [terminationGracePeriodSeconds](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution) as appropriate for your game server.
+
+Note that to maintain backward compatibility with Agones prior to the introduction of the `SafeToEvict` feature gate, if your game server previously configured the `cluster-autoscaler.kubernetes.io/safe-to-evict: true` annotation, we assume `eviction.safe: Always` is intended.
+
+### `safe: OnUpgrade`
+
+If your game server has a session length between ten minutes and one hour, you can set:
+
+```
+eviction:
+  safe: OnUpgrade
+```
+
+This will ensure the game server `Pod` can be evicted by node upgrades, but Cluster Autoscaler will not evict it.
+
+If you do not wish your game server to be disrupted by upgrades, you can either set `Never`, or, with a 10m-1h session length, set [terminationGracePeriodSeconds](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution) to one hour (or your session length), then intercept `SIGTERM` but do nothing with it. Either approach will allow your game server to run to completion.
+
+{{< alert title="GKE Autopilot" color="info" >}}
+GKE Autopilot supports only `Never` and `Always`, not `UpgradeOnly`.
+{{< /alert >}}
+
+## What's special about ten minutes and one hour?
+
+* **Ten minutes:** Cluster Autoscaler respects [ten minutes of graceful termination](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#does-ca-respect-gracefultermination-in-scale-down) on scale-down. On some cloud products, you can configure `--max-graceful-termination-sec` to change this, but it is not advised: Cluster Autoscaler is currently only capable of scaling down one node at a time, and larger graceful termination windows slow this down farther (see [autoscaler#5079](https://github.com/kubernetes/autoscaler/issues/5079)). If the ten minute limit does not apply to you, generally you should choose between `safe: Always` (for sessions less than an hour), or see [below](#considerations-for-long-sessions).
+
+* **One hour:** On many cloud products, `PodDisruptionBudget` can only block node upgrade evictions for a certain period of time - on GKE this is 1h. After that, the PDB is ignored, or the node upgrade fails with an error. Controlling `Pod` disruption for longer than one hour requires cluster configuration changes outside of Agones - see [below](#considerations-for-long-sessions).
```

Comment:
Do we know what the PBD timeout is on other cloud providers?

On GKE 1h is also the max time given for graceful termination before a pod is forcefully removed during node drain.

Reply:
From what I can tell gleaning other providers documentations, I think they err on the side of erroring rather than proceeding. I am still trying to research this.